src/Nethermind/Nethermind.State/IWorldState.cs

```diff
@@ -104,6 +104,8 @@ public interface IWorldState : IJournal<Snapshot>, IReadOnlyStateProvider
 
     void DecrementNonce(Address address) => DecrementNonce(address, UInt256.One);
 
+    void SetNonce(Address address, in UInt256 nonce);
```

Comment:
This is not a normal use-case method when interacting with `WorldState`. Might be useful on some setups, but not sure if should be just public on interface.